service/src/main/kotlin/fi/espoo/evaka/attendance/ChildAttendanceController.kt

```diff
@@ -276,23 +265,27 @@ class ChildAttendanceController(
                         Action.Unit.READ_CHILD_ATTENDANCES,
                         unitId
                     )
-                    val attendance = tx.getChildOngoingAttendance(childId, unitId)
-                    if (attendance == null) emptyList()
-                    else getPartialAbsenceThresholds(tx, clock, childId, unitId, attendance)
+                    val ongoingAttendance =
+                        tx.getChildOngoingAttendance(childId, unitId)
+                            ?: throw BadRequest("Cannot depart, has not yet arrived")
+                    val attendanceTimesToday =
+                        tx.getCompletedChildAttendanceTimes(childId, unitId, today) +
+                            ongoingAttendance.toTimeRange(HelsinkiDateTime.of(today, body.departed))
+                    getExpectedAbsenceCategories(
+                        tx = tx,
+                        date = today,
+                        childId = childId,
+                        presenceTimes = attendanceTimesToday
```

Comment:
Not probably a problem for this PR specially, but it seems that we now have two terms that mean the same thing: attendance and presence.